### PR TITLE
Add README for airflow

### PR DESCRIPTION
**Pull request by yangaws**
*Issue #, if available:*

*Description of changes:*
Add README for using SageMaker with Airflow.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have updated the [changelog](https://github.com/aws/sagemaker-python-sdk/blob/master/CHANGELOG.rst) with a description of my changes (if appropriate)
- [ ] I have updated any necessary [documentation](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/workflow/README.rst [hunk 1]
```diff
@@ -0,0 +1,162 @@
+=============================
+SageMaker Workflow in Airflow
```
Comment (yuanzhua) [inline src/sagemaker/workflow/README.rst]: Do you need to change the master readme and add a workflow section?

Comment (yangaws) [reply src/sagemaker/workflow/README.rst]: Good idea! One section added to master readme and linked to this one.

File: src/sagemaker/workflow/README.rst [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+Apache Airflow
+~~~~~~~~~~~~~~
+
+`Apache Airflow <https://airflow.apache.org/index.html>`_
+is a platform that enables you to programmatically author, schedule, and monitor workflows. Using Airflow,
+you can build a workflow for SageMaker training, hyperparameter tuning, batch transform and endpoint deployment.
+You can use any SageMaker deep learning framework or Amazon algorithms to perform above operations in Airflow.
+
+There are two ways to build SageMaker workflow. Using Airflow SageMaker operators or using Airflow PythonOperator.
```
Comment (eslesar-aws) [inline src/sagemaker/workflow/README.rst]: ...to build a SageMaker workflow.

File: src/sagemaker/workflow/README.rst [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+1. SageMaker Operators: Since Airflow 1.10.1, we contributed special operators just for SageMaker operations.
```
Comment (eslesar-aws) [inline src/sagemaker/workflow/README.rst]: In Airflow 1.10.1, the SageMaker team contributed special operators for SageMaker operations.

File: src/sagemaker/workflow/README.rst [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+Each operator takes a configuration dictionary that defines the corresponding operation. And we provide APIs to
```
Comment (eslesar-aws) [inline src/sagemaker/workflow/README.rst]: We provide APIs to generate the configuration dictionary in the SageMaker Python SDK. Currently, the following SageMaker operators are supported:

File: src/sagemaker/workflow/README.rst [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+generate the configuration dictionary in SageMaker Python SDK. Current supported SageMaker operators are:
+
+* ``SageMakerTrainingOperator``
+* ``SageMakerTuningOperator``
+* ``SageMakerModelOperator``
+* ``SageMakerTransformOperator``
+* ``SageMakerEndpointConfigOperator``
+* ``SageMakerEndpointOperator``
+
+2. PythonOperator: Airflow built-in operator that could execute Python callables. You could use SageMaker Python SDK to
```
Comment (eslesar-aws) [inline src/sagemaker/workflow/README.rst]: Airflow built-in operator that executes Python callables. You can use the PythonOperator to  execute operations in the SageMaker Python SDK to creat a SageMaker workflow.

File: src/sagemaker/workflow/README.rst [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+customize your workflow with PythonOperator.
+
+Using Airflow on AWS
+~~~~~~~~~~~~~~~~~~~~
+
+Turbine is an open source AWS CloudFormation template to create Airflow resources stack on AWS.
```
Comment (eslesar-aws) [inline src/sagemaker/workflow/README.rst]: Turbine is an open-source AWS CloudFormation template that enables you to create an Airflow resource stack on AWS.

File: src/sagemaker/workflow/README.rst [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+You can get it here: https://github.com/villasv/aws-airflow-stack
+
+Using Airflow SageMaker Operators
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Starting from Airflow 1.10.1, you could use SageMaker operators in Airflow. All SageMaker operators take a
```
Comment (eslesar-aws) [inline src/sagemaker/workflow/README.rst]: Starting wiht Airflow 1.10.1, you can use SageMaker operators in Airflow. All SageMaker operators take a configuration dictionary that can be generated by the SageMaker Python SDK. For example:

File: src/sagemaker/workflow/README.rst [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+                                   inputs=your_training_data_s3_uri)
+
+    # trans_config specifies SageMaker batch transform configuration
+    trans_config = transform_config_from_estimator(estimator=estimator,
+                                                   instance_count=1,
+                                                   instance_type='ml.m4.xlarge',
+                                                   data=your_transform_data_s3_uri,
+                                                   content_type='text/csv')
+
+Now we can pass these configurations to related SageMaker operators and create the workflow:
```
Comment (eslesar-aws) [inline src/sagemaker/workflow/README.rst]: Now you can pass these configurations to the corresponding SageMaker operators and create the workflow:

File: src/sagemaker/workflow/README.rst [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+        wait_for_completion=True,
+        dag=dag)
+
+    transform_op.set_upstream(train_op)
+
+Using Airflow Python Operator
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+`Airflow PythonOperator <https://airflow.apache.org/howto/operator.html?#pythonoperator>`_
+is a built-in operator that can execute any Python callables. If you want to build the SageMaker workflow in a more
```
Comment (eslesar-aws) [inline src/sagemaker/workflow/README.rst]: ...execute any Python callable.

File: src/sagemaker/workflow/README.rst [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+flexible way,  you could write your python callables for SageMaker operations using SageMaker Python SDK. For example:
```
Comment (eslesar-aws) [inline src/sagemaker/workflow/README.rst]: ...flexible way, writer your python callables for SageMaker operatoins by using the SageMaker Python SDK.

File: src/sagemaker/workflow/README.rst [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+        return estimator.latest_training_job.job_name
+
+    # callable for SageMaker batch transform
+    def transform(data, **context):
+        training_job = context['ti'].xcom_pull(task_ids='training')
+        estimator = TensorFlow.attach(training_job)
+        transformer = estimator.transformer(instance_count=1, instance_type='ml.c4.xlarge')
+        transformer.transform(data, content_type='text/csv')
+
+Then you could build your workflow using PythonOperator with Python callables defined above:
```
Comment (eslesar-aws) [inline src/sagemaker/workflow/README.rst]: Then build your workflow by using the PythonOperator with the Python callables defined above:

File: src/sagemaker/workflow/README.rst [hunk 12]
```diff
@@ -0,0 +1,162 @@
...
+    transform_op = PythonOperator(
+        task_id='transform',
+        python_callable=transform,
+        op_args=[transform_data_s3_uri],
+        provide_context=True,
+        dag=dag)
+
+    transform_op.set_upstream(train_op)
+
+A workflow with SageMaker training and batch transform is finished! In this way, you could customize your Python
```
Comment (eslesar-aws) [inline src/sagemaker/workflow/README.rst]: A workflow that runs a SageMaker training job and  a batch transform job is finished. You can customize your Python callables with the SageMaker Python SDK according to your needs, and build more flexible and powerful workflows.

File: src/sagemaker/workflow/README.rst (hunk 4 above)
Comment (yangaws) [reply src/sagemaker/workflow/README.rst]: Updated.

File: src/sagemaker/workflow/README.rst (hunk 2 above)
Comment (yangaws) [reply src/sagemaker/workflow/README.rst]: Updated.

File: src/sagemaker/workflow/README.rst (hunk 3 above)
Comment (yangaws) [reply src/sagemaker/workflow/README.rst]: Updated.

File: src/sagemaker/workflow/README.rst (hunk 5 above)
Comment (yangaws) [reply src/sagemaker/workflow/README.rst]: Updated.

File: src/sagemaker/workflow/README.rst (hunk 6 above)
Comment (yangaws) [reply src/sagemaker/workflow/README.rst]: Updated.

File: src/sagemaker/workflow/README.rst (hunk 7 above)
Comment (yangaws) [reply src/sagemaker/workflow/README.rst]: Updated.

File: src/sagemaker/workflow/README.rst (hunk 8 above)
Comment (yangaws) [reply src/sagemaker/workflow/README.rst]: Updated.

File: src/sagemaker/workflow/README.rst (hunk 9 above)
Comment (yangaws) [reply src/sagemaker/workflow/README.rst]: Updated.

File: src/sagemaker/workflow/README.rst (hunk 10 above)
Comment (yangaws) [reply src/sagemaker/workflow/README.rst]: Updated.

File: src/sagemaker/workflow/README.rst (hunk 11 above)
Comment (yangaws) [reply src/sagemaker/workflow/README.rst]: Updated.

File: src/sagemaker/workflow/README.rst (hunk 12 above)
Comment (yangaws) [reply src/sagemaker/workflow/README.rst]: Updated.

**Pull request merged**